Source/SwiftLintFramework/Rules/DiscouragedInitRule.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  DiscourageInitRule.swift
+//  SwiftLint
+//
+//  Created by Ornithologist Coder on 8/1/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct DiscouragedInitRule: ASTRule, OptInRule, ConfigurationProviderRule {
+    public var configuration = DiscouragedInitConfiguration()
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "discouraged_init",
+        name: "Discouraged Init",
+        description: "Discouraged direct initialization of class.",
+        kind: .lint,
+        nonTriggeringExamples: [
+            "let foo = UIDevice.current",
+            "let foo = Bundle.main",
+            "let foo = Bundle(path: \"bar\")",
+            "let foo = Bundle(identifier: \"bar\")"
+        ],
+        triggeringExamples: [
+            "↓UIDevice()",
+            "↓Bundle()",
+            "let foo = ↓UIDevice()",
+            "let foo = ↓Bundle()",
+            "let foo = bar(bundle: ↓Bundle(), device: ↓UIDevice())"
+        ]
+    )
+
+    public func validate(file: File,
+                         kind: SwiftExpressionKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        guard
+            let offset = dictionary.nameOffset,
+            let name = dictionary.name,
+            kind == .call,
+            dictionary.bodyLength == 0,
```

Comment:
so, this rule actually just works for inits without parameters, right?

In this case, I'd rename the rule and try to improve the description as well.